File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        while (s.length() > 2) {
            StringBuilder t = new StringBuilder();
            for (int i = 0; i < s.length() - 1; i++) {
                int x = ((s.charAt(i) - '0') + (s.charAt(i + 1) - '0')) % 10;
                t.append((char)(x + '0'));
            }
            s = t.toString();
        }
        return s.charAt(0) == s.charAt(1);
    }
}

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        while len(s) > 2:
            t = ""
            for i in range(len(s) - 1):
                x = (int(s[i]) + int(s[i + 1])) % 10
                t += str(x)
            s = t
        return s[0] == s[1]
